pkg/models/tenant/tenant.go

```diff
@@ -21,63 +21,91 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/klog"
 	"kubesphere.io/kubesphere/pkg/api"
 	iamv1alpha2 "kubesphere.io/kubesphere/pkg/apis/iam/v1alpha2"
 	tenantv1alpha1 "kubesphere.io/kubesphere/pkg/apis/tenant/v1alpha1"
+	"kubesphere.io/kubesphere/pkg/apiserver/authorization/authorizer"
+	"kubesphere.io/kubesphere/pkg/apiserver/authorization/authorizerfactory"
 	"kubesphere.io/kubesphere/pkg/informers"
 	"kubesphere.io/kubesphere/pkg/models/iam/am"
 	"kubesphere.io/kubesphere/pkg/simple/client/k8s"
 )
 
 type Interface interface {
-	ListWorkspaces(username string) (*api.ListResult, error)
-	ListNamespaces(username, workspace string) (*api.ListResult, error)
+	ListWorkspaces(user user.Info) (*api.ListResult, error)
+	ListNamespaces(user user.Info, workspace string) (*api.ListResult, error)
 }
 
 type tenantOperator struct {
-	informers informers.InformerFactory
-	am        am.AccessManagementInterface
+	informers  informers.InformerFactory
```

Comment:
if only kubesphereinformer needed,  it's unnecessary to import the whole InformerFactory

Reply:
kubernetesinformer is also used